tests/geospatial/test_netcdf_to_zarr.py

```diff
@@ -0,0 +1,93 @@
+import xarray as xr
+from dask.utils import format_bytes
+
+
+def test_netcdf_to_zarr(
+    scale,
+    s3,
+    s3_url,
+    client_factory,
+    cluster_kwargs={
+        "workspace": "dask-engineering",
+        "region": "us-west-2",
+        "wait_for_workers": True,
+    },
+    scale_kwargs={
+        "small": {"n_workers": 10},
+        "medium": {"n_workers": 100},
+        "large": {"n_workers": 200},
+    },
+):
+    with client_factory(
+        **scale_kwargs[scale], **cluster_kwargs
+    ) as client:  # noqa: F841
+        # Define models and variables of interest
+        models = [
+            "ACCESS-CM2",
+            "ACCESS-ESM1-5",
+            "CMCC-ESM2",
+            "CNRM-CM6-1",
+            "CNRM-ESM2-1",
+            "CanESM5",
+            "EC-Earth3",
+            "EC-Earth3-Veg-LR",
+            "FGOALS-g3",
+            "GFDL-ESM4",
+            "GISS-E2-1-G",
+            "INM-CM4-8",
+            "INM-CM5-0",
+            "KACE-1-0-G",
+            "MIROC-ES2L",
+            "MPI-ESM1-2-HR",
+            "MPI-ESM1-2-LR",
+            "MRI-ESM2-0",
+            "NorESM2-LM",
+            "NorESM2-MM",
+            "TaiESM1",
+            "UKESM1-0-LL",
+        ]
+        variables = [
+            "hurs",
+            "huss",
+            "pr",
+            "rlds",
+            "rsds",
+            "sfcWind",
+            "tas",
+            "tasmax",
+            "tasmin",
+        ]
+
+        if scale == "small":
+            # 130 files (152.83 GiB)
+            # One model and one variable
+            models = models[:1]
+            variables = variables[:1]
+        elif scale == "medium":
+            # 715 files (XX TiB)
+            # One model and all variables
+            models = models[:1]
+        else:
+            # 11635 files (XX TiB)
+            # All models and variables
+            pass
+
+        # Get netCDF data files -- see https://registry.opendata.aws/nex-gddp-cmip6
+        # for dataset details.
+        file_list = []
+        for model in models:
+            for variable in variables:
+                source_directory = f"s3://nex-gddp-cmip6/NEX-GDDP-CMIP6/{model}/historical/r1i1p1f1/{variable}/*.nc"
+                file_list += [f"s3://{path}" for path in s3.glob(source_directory)]
+        files = [s3.open(f) for f in file_list]
+        print(f"Processing {len(files)} NetCDF files")
+
+        ds = xr.open_mfdataset(
+            files,
+            engine="h5netcdf",
+            combine="nested",
+            concat_dim="time",
+            parallel=True,
+        )
+        print(f"Converting {format_bytes(ds.nbytes)} from NetCDF to Zarr")
+        ds.to_zarr(s3_url)
```

Comment:
@maxrjones thanks for pointing to this dataset over in https://github.com/coiled/benchmarks/discussions/1545#discussioncomment-10681943. Does this looks like what you've seen in the wild?

Reply:
I've gone ahead and added a rechunking step (from "pancake" to "pencil" chunks), which seems to be pretty common when cloud-optimizing a NetCDF dataset to Zarr

Reply:
Yeah, I think it's right that when cloud-optimizing it's usually best to include a rechunking step. But, there's many cases in which people avoid the cloud-optimization step and do subsequent analyses on the data loaded directly from the original NetCDF files.

Reply:
> Yeah, I think it's right that when cloud-optimizing it's usually best to include a rechunking step.

Awesome, thanks for confirming that's usually the case 👍 

> But, there's many cases in which people avoid the cloud-optimization step and do subsequent analyses on the data loaded directly from the original NetCDF files

I guess in this case, the "subsequent analyses" is just writing to Zarr. Do you think this still captures user pain well? FWIW my experience is using `xr.open_mfdataset(..., parallel=True)` + any other step performs very poorly

Reply:
> I guess in this case, the "subsequent analyses" is just writing to Zarr. Do you think this still captures user pain well? FWIW my experience is using xr.open_mfdataset(..., parallel=True) + any other step performs very poorly

Yeah, it could make sense to focus first on the simplest operations that perform poorly. You mention that going from pancake to pencil chunks is a common step that performs poorly. That's true and is the motivation for the rechunker library. The worst performance would likely be seen when combining operations that perform optimally on pancake oriented chunks with operations that perform optimally on churro oriented chunks.

Reply:
> pancake oriented chunks...churro oriented chunks

💯